src/libs/ascent/runtimes/expressions/ascent_conduit_reductions.cpp

```diff
@@ -481,6 +594,67 @@ struct SumFunctor
   }
 };
 
+struct DFAddFunctor
+{
+  template<typename T, typename Exec>
+  void operator()(const MemoryAccessor<T> l_accessor,
+  			   const MemoryAccessor<T> r_accessor,
+  			   MemoryAccessor<T> output,
+                           const Exec &) const
+  {
+    const int l_size = l_accessor.m_size;
+    const int r_size = r_accessor.m_size;
+    bool diff_sizes = false;
+    const int size; 
+    const int max_size;
+
+    size = max_size = l_size; 
+    if(l_size != r_size)
+    {
+      size = min(l_size, r_size);
+      max_size = max(l_size, r_size);
+      diff_sizes = true;
+    }
+
+    double values[max_size];
+
+    using for_policy = typename Exec::for_policy;
+    using reduce_policy = typename Exec::reduce_policy;
+
+    ascent::forall<for_policy>(0, size, [=] ASCENT_LAMBDA(index_t i)
+    {
+      const T val = l_accessor[i] + r_accessor[i];
+      values[i] = val;
+    });
+    ASCENT_DEVICE_ERROR_CHECK();
+
+    if(diff_sizes)
+    {
+      if(l_size > r_size)
+      {
+        ascent::forall<for_policy>(size, l_size, [=] ASCENT_LAMBDA(index_t i)
+        {
+          const T val = l_accessor[i];
+          values[i] = val;
+        });
+        ASCENT_DEVICE_ERROR_CHECK();
```

Comment:
to confirm my understanding:

If one field is larger than another, the output will be sized to the larger field and the remaining vals are simply copied.

Reply:
Correct. That's what I was going for here, plus zero so simply copied what's extra. I figures it's ok to take them for different sizes? Or do I need to be concerned about topology further down the pipeline?

Reply:
Thanks, yes this makes sense. 

We aren't likely to hit these cases often b/c Blueprint fields on the same topology with the same assoc should share cardinality. But its much better to handle the logic vs having a real head scratcher crash down the line.